Assets/Scripts/Interactables/Interactable.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Application;
+using UnityEngine;
+
+public class Interactable : MonoBehaviour, IInteractable
+{
+    private Animator interactableAnim;
+    private bool interactableOn = false;
+
+    public delegate void OnInteractEventHandler(Interactable source);
+    public event OnInteractEventHandler OnInteractEnd;
+
+
+    // Start is called before the first frame update
+    protected void Start()
+    {
+        interactableAnim = GetComponentInChildren<Animator>();
+    }
+
+ 
+    protected void OnTriggerEnter(Collider other)
+    {
+        if (other.gameObject.layer == LayerMask.NameToLayer("PLAYER"))
+        {
+            if (!interactableOn)
+            {
+                ShowInteractUI();
+            }
+        }
+    }
+
+
+    protected void OnTriggerExit(Collider other)
+    {
+        if (other.gameObject.layer == LayerMask.NameToLayer("PLAYER"))
+        {
+            HideInteractUI();
+        }
+    }
+
+
+    protected void OnTriggerStay(Collider other)
+    {
+        // Show interactable button in the UI
+
+        //
```

Comment:
Did you mean to write a comment here? Otherwise maybe just remove this line / fix the spacing here